transforms.go

```diff
@@ -97,6 +117,29 @@ func goSealTicket(src C.sector_builder_ffi_FFISealTicket) SealTicket {
 	}
 }
 
+func goCandidates(src *C.sector_builder_ffi_FFICandidate, size C.size_t) ([]Candidate, error) {
+	candidates := make([]Candidate, size)
+	if src == nil || size == 0 {
+		return candidates, nil
+	}
+
+	ptrs := (*[1 << 30]C.sector_builder_ffi_FFICandidate)(unsafe.Pointer(src))[:size:size]
```

Comment:
why are we casting this pointer as a gigabyte sized fixed array?

Reply:
Not sure tbh, I copied this from the other casts. cc @laser

Reply:
Not sure tbh, I copied this from the other casts. cc @laser

Reply:
See the [CGO Wiki](https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices) for an explanation, if ye so desire.